Binary files cowpatty-4.6/.cowpatty.c.swp and cowpatty-4.6-fixup/.cowpatty.c.swp differ
diff -uNr cowpatty-4.6/cowpatty.c cowpatty-4.6-fixup/cowpatty.c
--- cowpatty-4.6/cowpatty.c	2009-07-03 12:27:07.361217549 -0700
+++ cowpatty-4.6-fixup/cowpatty.c	2009-07-03 13:36:42.657261185 -0700
@@ -455,17 +455,36 @@
 		return;
 	}
 
-	if (cdata->ver == WPA_KEY_INFO_TYPE_HMAC_MD5_RC4) {
-		/* Check for WPA key, and pairwise key type */
-		if (eapolkeyhdr->type != 254 || 
-				(key_info & WPA_KEY_INFO_KEY_TYPE) == 0) {
-			return;
-		}
-	} else if (cdata->ver == WPA_KEY_INFO_TYPE_HMAC_SHA1_AES) {
-		if (eapolkeyhdr->type != 2 ||
-				(key_info & WPA_KEY_INFO_KEY_TYPE) == 0) {
-			return;
+	if (opt->nonstrict == 0) {
+
+		if (cdata->ver == WPA_KEY_INFO_TYPE_HMAC_MD5_RC4) {
+			/* Check for WPA key, and pairwise key type */
+			if (eapolkeyhdr->type != 254 || 
+					(key_info & WPA_KEY_INFO_KEY_TYPE) == 0) {
+				return;
+			}
+		} else if (cdata->ver == WPA_KEY_INFO_TYPE_HMAC_SHA1_AES) {
+			if (eapolkeyhdr->type != 2 ||
+					(key_info & WPA_KEY_INFO_KEY_TYPE) == 0) {
+				return;
+			}
+		}
+
+	} else {
+
+                if (cdata->ver == WPA_KEY_INFO_TYPE_HMAC_MD5_RC4) {
+                        /* Check for WPA key, and pairwise key type */
+                        if ((eapolkeyhdr->type != 2 && eapolkeyhdr->type != 254) ||
+                                        (key_info & WPA_KEY_INFO_KEY_TYPE) == 0) {
+                                return;
+                        }
+                } else if (cdata->ver == WPA_KEY_INFO_TYPE_HMAC_SHA1_AES) {
+                        if ((eapolkeyhdr->type != 2 && eapolkeyhdr->type != 254) ||
+                                        (key_info & WPA_KEY_INFO_KEY_TYPE) == 0) {
+                                return;
+                        }
 		}
+
 	}
 
 	if (opt->nonstrict == 0) {
